lib/JSON/Validator.pm

```diff
@@ -180,9 +180,12 @@ sub _load_schema_from_url {
   $tx = $self->ua->get($url);
   die $tx->error->{message} if $tx->error;
 
-  if ($cache_dir and -w $cache_dir) {
-    $cache_file = File::Spec->catfile($cache_dir, $cache_file);
-    Mojo::Util::spurt($tx->res->body, $cache_file);
+  for my $cache_dir (@{$self->cache_paths}) {
+    if ($cache_dir and -w $cache_dir and $cache_dir ne $DEFAULT_CACHE_DIR) {
+      $cache_file = File::Spec->catfile($cache_dir, $cache_file);
+      Mojo::Util::spurt($tx->res->body, $cache_file);
+      last;
```

Comment:
Not 100% sure about the last here - do we want to cache specs in *all* cache dirs?